Common/cpp/Tools/JSScheduler.cpp

```diff
@@ -5,8 +5,15 @@
 namespace reanimated {
 
 void JSScheduler::scheduleOnJS(std::function<void(jsi::Runtime &rt)> job) {
-  jsCallInvoker_->invokeAsync(
-      [job = std::move(job), &rt = rnRuntime_] { job(rt); });
+  if (runtimeExecutor_ != nullptr) {
+    runtimeExecutor_(
+        [job = std::move(job)](jsi::Runtime &runtime) { job(runtime); });
```

Comment:
I confirm that this callback is indeed invoked on the JS thread:

<img width="989" alt="Screenshot 2024-03-28 at 10 24 09" src="https://github.com/software-mansion/react-native-reanimated/assets/20516055/51320e97-097d-47fa-980c-d5aa2adef985">